content/blog/getting-started-with-hpe-greenlake-api-for-backup-and-recovery.md

```diff
@@ -0,0 +1,591 @@
+
+The below list detailed the required steps:
+
+1. I figured out the protection-job-id that is associated with `“0-Linux-Demo-VM02”`. and the  cloud backup schedule Id of the virtual machine. To achieve that, I used the HPE GreenLake [API](`https://developer.greenlake.hpe.com/docs/greenlake/services/backup-recovery/public/openapi/backup-recovery-public-v1beta1/operation/DataManagementJobsList`) for Backup and Recovery `GET /backup-recovery/v1beta1/protection-jobs` and `filter assetInfo/id eq "{VM-id}”` as shown below. Note that the variable `{vmId}` contained the value of the virtual machine id as discovered in previous step, namely `"<virtual-machine-id>"`. The response body’s JSON structure contained the id of the protection job associated with `“0-Linux-Demo-VM02”`. From the same response body, I recognized that cloud protection is the `scheduleId no 3`. The API used for this:
+
+   ```shellsession
+   GET /backup-recovery/v1beta1/protection-jobs?filter=assetInfo/id eq {{vmId}}&select=assetInfo,id,operational,protections
+   ```
+
+![API to figure out protection-jobs](/img/api-to-figure-out-protection-jobs.png)
+
+2. This use case is equivalent to run now button at the schedule of the `“0-Linux-Demo-V02”` on the selected cloud protection as shown in below figure. After clicking `“Run Now”` button, a cloud protection will commence against the virtual machine, a recovery point is going to be created at the cloud protection store. On next step, I will invoke `"Run Now"` on the `HPE Cloud Protection Store` using HPE GreenLake API for Backup and Recovery.
+
+![UI for run now protection jobs](/img/gui-run-now-cloud-protection.png)
+
+3. The GreenLake API to accomplish the use case above was `POST /backup-recovery/v1beta1/protection-jobs/:id/run`, and the documentation of this [API](https://developer.greenlake.hpe.com/docs/greenlake/services/backup-recovery/public/openapi/backup-recovery-public-v1beta1/operation/DataManagementJobRun) also list the required JSON structure for the request body. I created a JSON request body structure as the following:
+
+   ```jsonc
+   {
+      "﻿fullBackup": false,
+      "﻿ScheduleIds": [3]
+   }
+   ```
+
+> **Note** that there was a key called `“fullBackup”` inside the JSON request body to enable the creation of full protection where a backup will be created independently from the existing copies in the protection store. I also entered number 3 into `ScheduleIds` JSON array, to represent the cloud backup schedule. The below figure shows an example of the execution of `"run now"` without full backup protection of the third schedule which is cloud protection of this virtual machine. The value `<protection-jobs-id>` will be entered from the parameter of this API in this manner: `POST /backup-recovery/v1beta1/protection-jobs/"<protection-jobs-id>"/run`.
```

Comment:
Take out the word "that" and just say "Note: There was a key..."
Add a comma prior to "which is cloud protection" in the second to last sentence.